docs/reference/search/search_context.asciidoc

```diff
@@ -0,0 +1,92 @@
+[[search-context]]
+==== Search Context
+
+By default, a search request executes against the latest point in time readers of the
+target indices. Sometimes it's more preferred to execute multiple search requests using
+the same point in time readers. For example, the combined result of the initial and
+subsequent search_after requests would be more consistent if they use the same point
+in time readers. This can be done by using a single search context for all search requests.
+
+A search context must be opened in a separate step before being used in subsequent
+search requests. The keep_alive parameter tells Elasticsearch how long it should keep
+the seach context alive, e.g. `?keep_alive=5m`.
+
+[source,console]
+--------------------------------------------------
+POST /twitter/_search_context?keep_alive=1m
+--------------------------------------------------
+// TEST[setup:twitter]
+
+The result from the above request includes a `id`, which should
+be passed to the `id` of the `search_context` parameter of a search request.
+
+POST /_search <1>
+{
+    "size": 100,
+    "query": {
+        "match" : {
+            "title" : "elasticsearch"
+        }
+    },
+    "search_context": {
+	    "id":  "46ToAwEHdHdpdHRlchZSa1dRZDFELVNLbTZkQVJudUtGMFFnAAAWNVAwS09JWTdTRUdibWE1ZC1id0tjQRRURnhPWjNFQmI3bmVOeVZReS1tRAAAAAAAAAAB", <2>
+	    "keep_alive": "1m"  <3>
+    }
+}
+
+<1> A search request with `search_context` must not specify `index`, `routing`,
```

Comment:
Does it mean that this search request will be forwarded to specific shards?  
If yes, what happens if these shards are no longer avaialable?

Reply:
Thanks for checking, Mayya. The request will fail if some shards are not available (depending on the partial result flag). The scroll API has the same limitation. However, as the search context becomes stateless, we will work a follow-up to replace the unavailable shards with other copies when possible.